### DISCUSSION TITLE
... is not registered in TypeRegistry, what does it mean?

### DISCUSSION THREAD

**Discussion by stanley355** [jakobhellermann/bevy-inspector-egui#130] [Q&A]
![image](https://user-images.githubusercontent.com/53996155/227760933-4556219f-3a31-4613-b66e-492868690ee3.png)

I always found something is not registered in TypeRegistry on the egui, what does it mean?

**Comment by jakobhellermann** [comment]
When you have a custom type, the inspector needs to use the `Reflect` trait to show UI for it.

To do that you need to `#[derive(Reflect)]` and call `app.register_type::<PlayerSpriteIndices>()`. Maybe the message could explicitly call that out.

**Reply by harderj** [reply]
Hi, I am trying to do exactly this for a struct which has types from another library (nalgebra):
![image](https://github.com/jakobhellermann/bevy-inspector-egui/assets/5573365/03fa73f4-dc66-41e5-b98e-25c1b3e249ef)
So I guess I need to implement type_path (and possibly more?) for nalgebra::Vector2, but I am a bit lost as to how, maybe someone here can give me hint :crossed_fingers:
